prac_05/wimbledon.py

```diff
@@ -0,0 +1,53 @@
+"""
+Emma Naumann
+CP1404 Practical 5
+Estimate: 30 minutes
+Actual:
+"""
+
+FILENAME = "wimbledon.csv"
+INDEX_COUNTRY = 1
+INDEX_CHAMPION = 2
+
+
+def main():
+    """Read data file and print details about Wimbledon champions and countries."""
+    records = get_records(FILENAME)
+    champion_to_count, countries = process_records(records)
+    display_results(champion_to_count, countries)
+
+
+def get_records(filename):
+    """Get records from file and store as a list of lists."""
+    records = []
+    with open(FILENAME, "r", encoding="utf-8-sig") as in_file:
+        in_file.readline()
+        for line in in_file:
+            parts = line.strip().split(",")
+            records.append(parts)
+    return records
+
+
+def process_records(records):
+    """Create dictionary of champions and set of countries from records."""
+    champion_to_count = {}
+    countries = set()
+    for record in records:
+        countries.add(record[INDEX_COUNTRY])
+        try:
```

Comment:
Just a note that there are two alternatives way to do this if you wanted to. Both are in the lecture material slides and explained in the videos. The third way of the three total ways is shorter if you wanted to explore that option :)